drivers/scheduler/openshift/openshift.go

```diff
@@ -330,13 +332,66 @@ spec:
 	return err
 }
 
+// getImageSha get Imgae sha
+func getImageSha(ocpVersion string) (string, error) {
+	downloadURL := fmt.Sprintf("%s/%s/%s", OpenshiftMirror,
+		ocpVersion, mdFileName)
+	request := netutil.HttpRequest{
+		Method:   "GET",
+		Url:      downloadURL,
+		Content:  "application/json",
+		Body:     nil,
+		Insecure: true,
+	}
+	log.Infof("URL %s", downloadURL)
+	content, err := netutil.DoRequest(request)
+	if err != nil {
+		log.Errorf("Failed to get Get content from  %s, error %v", downloadURL, err)
+		return "", err
+	}
+	//Convert the body to type string
+	contentInString := string(content)
+	parts := strings.Split(contentInString, "\n")
+	for _, a := range parts {
+		if strings.Contains(a, "Image Digest:") {
+			return strings.Split(a, "`")[1], nil
+		}
+	}
+	return "", fmt.Errorf("Failed to find Image sha: in  %s", downloadURL)
+}
+
 func startUpgrade(upgradeVersion string) error {
 	var output []byte
 	var err error
-
+	var shaName string
 	args := []string{"adm", "upgrade", fmt.Sprintf("--to=%s", upgradeVersion)}
 	t := func() (interface{}, bool, error) {
 		if output, err = exec.Command("oc", args...).CombinedOutput(); err != nil {
+			// If OC upgrade failed for reasons like not recommended upgrade or path does not exist.
+			forceUpgrade := "specify --to-image"
+			notRecommended := "is not one of the recommended updates, but is available"
```

Comment:
nit: Might want to grep for specific words that will always come up in not recommended upgrades, looking for a long sentence might not be a case for all Openshift versions, but I don't know for sure, in my opinion the smaller the string you are looking for the better

Reply:
> nit: Might want to grep for specific words that will always come up in not recommended upgrades, looking for a long sentence might not be a case for all Openshift versions, but I don't know for sure, in my opinion the smaller the string you are looking for the better
Tried to check specific words , but I feel keeping this  long sentence will help to improve readability of the code.